_minutes/2025-07-31-wecg.md

```diff
@@ -0,0 +1,172 @@
+
+ * [simeon] TPAC registrations are open. The meeting schedule is also published; we have 12 hours of meeting time split over Monday and Thursday.
+
+[Issue 849](https://github.com/w3c/webextensions/issues/849): Inconsistency: native messaging size limit
+
+ * [simeon] Didn't we discuss this last time?
+ * [rob] Chrome's perspective was missing; also in the meantime additional discussion happened on the issue.
+ * [simeon] The author is asking for browsers to provide chunking so that native messaging limits are not exposed to extensions.
+ * [rob] That's an assumption on the implementation, and not necessarily the technical reason.
+ * [simeon] They'd like a higher limit.
+ * [rob] 1 MB can be higher, but what limit would be good enough?
+ * [devlin] Heard of use case like JSON-encoding a video, in which case 64 MB is too small. Realistic use cases also include password stores, which could be large, and 1 MB would be too small. We could bump to 64 MB, and also have other ways to overcome IPC limits. But as the native messaging implementation is not optimal, we're concerned about the impact. And it also discourages. (..) Recommend using localhost instead.
+ * [rob] Aren't browsers locking down localhost access?
+ * [devlin] Arguably, but this is in extensions. On the web the ability to run a native application is a sandbox escape, but it's acceptable in extensions.
+ * [rob] Requires a server to be continuously running, a bit different than dropping a file in a location and letting the browser launch it.
+ * [devlin] Mostly just trying to say that I'm not convinced increasing the native messaging limit is the right path.
+ * [rob] Are we willing to raise the limit from 1 MB (native app-to-browser) and 64 MB (browser-to-native app).
+ * [devlin] Hesitant to raise beyond 64MB because copying string buffers larger than that is nontrivial and has performance implications.
+ * [rob] 64MB is a reasonable limit from extensions to native messaging host, but the opposite direction is 1MB. Are we in favor of raising that limit to 64MB?
+ * [devlin] Have to look into why we are at 1 MB, but in principle amendable to raising the limit to 64 MB.
+ * [kiara] 64 MB in either direction works for Safari, as long as 80 MB is not exceeded (due to xpc service limits).
+ * **Consensus**: alignment on raising limit to 64 MB in either direction.
+ * [rob] Do we want to set an environment variable or something to expose limits to native applications?
+ * [tomislav] They can always test the limits manually, so there's no reason to hide this variable.
+
+[Issue 855](https://github.com/w3c/webextensions/issues/855): Proposal: Clarify the scale parameter for tabs.captureVisibleTab
+
+ * [devlin] We added “rect” for visible reason, and noticed that “scale” does something different. Our implementation right now () uses “scale” to scale from device independent pixels to pixels, opposed to actually upscale the captured region.
```

Comment:
I'm not 100% sure this is what was actually said, but it makes more sense to me than the current text.

```suggestion
 * [devlin] We added “rect” for a visible region, and noticed that “scale” does something different. Our implementation right now () uses “scale” to scale from device independent pixels to pixels, opposed to actually upscale the captured region.
```